src/main/scala/org/openhorizon/exchangeapi/route/user/User.scala

```diff
@@ -87,19 +95,35 @@ trait User extends JacksonSupport with AuthenticationSupport {
             UsersTQ.getUserIfAdmin(realResource)
           else
             UsersTQ.getUser(realResource)
-        db.run(query.result).map({
-          list =>
-            logger.debug(s"GET /orgs/$organization/users/$realUsername result size: ${list.size}")
-            
-            val users: Map[String, org.openhorizon.exchangeapi.table.user.User] =
-              list.map(e => e.username -> UserTable(if (identity.isSuperUser || identity.isHubAdmin) e.hashedPw else StrConstants.hiddenPw, e.admin, e.hubAdmin, e.email, e.lastUpdated, e.updatedBy)).toMap
-            val code: StatusCode =
-              if (users.nonEmpty)
-                StatusCodes.OK
-              else
-                StatusCodes.NotFound
-            (code, GetUsersResponse(users, 0))
-        })
+      db.run(query.result).flatMap { list =>
+        if (list.nonEmpty) {
+        val userRow = list.head
+
+        db.run(ApiKeysTQ.getByUser(userRow.username).result).map { keys =>
+        val keyMetadata = keys.map(row =>
+          ApiKeyMetadata(
+          id = row.id,
+          description = row.description,
+          owner = null, 
+          lastUpdated = row.modifiedBy ))
+
+        val user = org.openhorizon.exchangeapi.table.user.User(
+        password = if (identity.isSuperUser || identity.isHubAdmin) userRow.hashedPw else StrConstants.hiddenPw,
+        admin = userRow.admin,
+        hubAdmin = userRow.hubAdmin,
+        email = userRow.email,
+        lastUpdated = userRow.lastUpdated,
+        updatedBy = userRow.updatedBy,
+        apikeys = Some(keyMetadata)
+      )
+
+      val usersMap = Map(userRow.username -> user)
+      (StatusCodes.OK, GetUsersResponse(usersMap, 0))
+    }
+  } else {
+      Future.successful((StatusCodes.NotFound, GetUsersResponse(Map.empty, 0)))
```

Comment:
The `Complete` directive handles this, you do not have to wrap this in a Future. its implicit.